proxy/http/HttpBodyFactory.cc

```diff
@@ -73,6 +73,16 @@ HttpBodyFactory::fabricate_with_old_api(const char *type, HttpTransact::State *c
   bool found_requested_template = false;
   bool plain_flag               = false;
 
+  ///////////////////////////////////////////////
+  // if suppressing this response, return NULL //
+  ///////////////////////////////////////////////
+  if (is_response_suppressed(context)) {
```

Comment:
This implementation of this function has a header commet that says "LOCKING: must be called with lock taken" (although it's not clear why).

Reply:
Yeah, I think it’s somewhat of a poorer implementation of supporting dynamic reconfiguration of the error templates. The locks block the event loops which has been a PITA. We need a refactor of this stuff but that’s outside the scope of what this PR is attempting to do.

FWIW, this PR at least makes a small optimization by avoiding locking and malloc when error pages are suppressed.

Reply:
@ywkaras Since you seem to have reviewed the PR, could you give a +1 if you are satisfied :) ?

Reply:
Won't help I'm not a committer.  I don't really understand the multi-thread access strategy for this class (or generally in TS).